mllib-local/src/main/scala/org/apache/spark/mllib/util/NumericParser.scala

```diff
@@ -21,16 +21,14 @@ import java.util.StringTokenizer
 
 import scala.collection.mutable.{ArrayBuilder, ListBuffer}
 
-import org.apache.spark.SparkException
-
 /**
  * Simple parser for a numeric structure consisting of three types:
  *
  *  - number: a double in Java's floating number format
  *  - array: an array of numbers stored as `[v0,v1,...,vn]`
  *  - tuple: a list of numbers, arrays, or tuples stored as `(...)`
  */
-private[mllib] object NumericParser {
+private[spark] object NumericParser {
```

Comment:
We can leave this file in mllib and remove the parse methods under `ml.linalg`, which was introduced very earlier and now basically replaced by UDTs.